File: Two_sum/Two_sum.cpp
//Solution 1 | Brute force

#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {

        int size = nums.size();

        for (int i=0; i<size; i++){
            for(int j = i + 1; j<size; j++){

                if(nums[i] + nums[j] == target){
                    return {i, j};
                }
            }
        }
        return {};
    }
};

int main() {
    // Create a Solution class Object
    Solution solution;

     // Vector & Target Definition
    std::vector<int> nums = {3, 2, 4};
    int target = 6;

     // Twosum function call
    std::vector<int> result = solution.twoSum(nums, target);

    // Result display
    if (!result.empty()) {
        std::cout << "Output: [" << result[0] << ", " << result[1] << "]" << std::endl;
    } else {
        std::cout << "There is no possible combination" << std::endl;
    }

    return 0;
}

File: Neetcode/Arrays/03-Two_sum/Two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        numap = {}
        
        for i, num in enumerate(nums):
            
            z = target - num

            if z in numap:
                return [numap[z], i] # [numap[z] == j, i]          
            numap[num] = i

        return []

nums=[1,3,4,2]
target=6

Solution = Solution()
print(Solution.twoSum(nums, target))